core/src/main/java/com/microsoft/applicationinsights/internal/agent/CoreAgentNotificationsHandler.java

```diff
@@ -177,6 +179,22 @@ public void httpMethodFinished(String identifier, String method, String correlat
         telemetryClient.track(telemetry);
     }
 
+    /**
+     * This is used to create Target string to be set in the RDD Telemetry
+     * According to spec, we do not include port 80 and 443 in target
+     * @param uriObject
+     * @return
+     */
+    private String createTarget(URI uriObject) {
+        assert uriObject != null;
+        String target = uriObject.getHost();
+        if (uriObject.getPort() != 80 && uriObject.getPort() != 443) {
+            target += ":" + uriObject.getPort();
+        }
+        target += " | " + target;
```

Comment:
@lmolkova there is a bug here. I am trying to append the same target, which is local string variable instead of passing the one from the httpFinished() method. I realized when doing final E2E testing. Will fix it, update PR and merge it.

Reply:
Fixed. Now it's production ready all good :)